src/envoy/utils/http_header_utils.cc

```diff
@@ -38,13 +38,17 @@ absl::string_view extractHeader(const Envoy::Http::HeaderMap& headers,
   return readHeaderEntry(entry);
 }
 
-absl::string_view getRequestHTTPMethodWithOverride(
-    absl::string_view originalMethod, const Envoy::Http::HeaderMap& headers) {
+bool handleHttpMethodOverride(Envoy::Http::HeaderMap& headers) {
   const auto* entry = headers.get(kHttpMethodOverrideHeader);
   if (entry) {
-    return entry->value().getStringView();
+    // Move the header.
```

Comment:
add a debug log entry here

Reply:
Done. One complication is that this function is outside a filter. We use `ENVOY_LOG_MISC` instead.